go/host/node/node.go

```diff
@@ -607,11 +606,37 @@ func (a *Node) storeBlockProcessingResult(result common.BlockSubmissionResponse)
 	}
 }
 
-// Distributes logs to subscribed clients.
+// Distributes logs to subscribed clients. We only send logs for rollups the subscription hasn't seen before.
 func (a *Node) sendLogsToSubscribers(result common.BlockSubmissionResponse) {
+	latestSeenRollupByID := map[rpc.ID]uint64{}
+
 	for subscriptionID, encLogsByRollup := range result.SubscribedLogs {
-		for _, encryptedLogs := range encLogsByRollup {
-			a.logsChs[subscriptionID] <- encryptedLogs
+		logSub, found := a.logSubscriptions[subscriptionID]
+		if !found {
+			log.Error("received a log for subscription with ID %s, but no such subscription exists", subscriptionID)
+			continue
+		}
+
+		for rollupNumber, encryptedLogs := range encLogsByRollup {
+			// We have received a log from a rollup this subscription hasn't seen before.
+			if rollupNumber > logSub.latestSeenRollup {
```

Comment:
Bit confused by this latestSeenRollup thing, why do we send logs from the enclave for a subscription when that log had already been sent to the host? Or is that maybe an optimisation to come later?

Reply:
I think it could be optimised away in the future. Don't know if there's a concern about the host dropping logs or missing a certain block submission response, and needing them again.